### PR TITLE
[css-multicol-2] Add ::column pseudo, per WG resolution. #10715

### PR DESCRIPTION
**Pull request by tabatkins**
@frivoal @rachelandrew Hi y'all, just drafting the ::column pseudo per #10715. Feel free to review/merge. I can also move the new section elsewhere if desired (or, potentially, to the Pseudo spec instead, but I feel like this is the better spot for it).

### CODE REVIEW

File: css-multicol-2/Overview.bs [hunk 1]
```diff
@@ -82,6 +82,48 @@ Stacking context</h3>
...
+Each ''::column'' has the same size and position
+as its corresponding column:
+the same size as the column's [=available space=] in the [=inline axis=]
+and the container's [=content box=] in the [=block axis=].
+(It does not cover the gaps/rules between columns.)
+
+''::column'' pseudo-elements are treated as children of their [=multi-column container=],
+without any contents.
+They <em>do not</em> wrap the contents of the column,
+just fill the same space.
```
Comment (fantasai) [inline css-multicol-2/Overview.bs:107]: I'm not sure this is true. I think we might reasonably say that they *do* wrap the contents of the column; and in the far future when we allow fragment-based styling, we might want to.

Comment (tabatkins) [reply css-multicol-2/Overview.bs:107]: We very specifically resolved that ::column was a leaf node, with no contents. See <https://github.com/w3c/csswg-drafts/issues/10715#issuecomment-2379924982> You even made some comments in support of that. ^_^

**Comment by tabatkins** [comment]
Do we want to add 'scroll-behavior' to the list of supported ::column properties? It's the only thing checked on the element by the [perform a scroll](https://drafts.csswg.org/cssom-view/#perform-a-scroll) algorithm, just to see if the scroll should be smooth or not. (While you can't (yet) scroll to a ::column via JS, the same algo is invoked when you click the ::scroll-marker associated with the ::column.)

**Comment by rachelandrew** [comment]
I think here is as good a place as any for it for now, we could always move it later. Unless @frivoal has concerns I'm happy to merge it here.

**Comment by AmeliaBR** [comment]
Is there an open issue for discussing which other properties might apply to a `::column` pseudo-element? As suggested in:

> Note: This list of properties and pseudo-elements that work on [::column](https://drafts.csswg.org/css-multicol-2/#selectordef-column) will likely be expanded in the future, as we develop more features that could usefully care about the position of a column without caring about the contents.

For example, I think it would be safe to add a background color & maybe full backgrounds, if someone wanted the column area to be coloured but not the gaps. Although then there might be a desire to add padding around the text but inside the coloured area, and... yeah, that's why there should probably be a standing issue (linked from that note) where you can start scoping out what is or isn't possible for this psuedo-element.

**Comment by rachelandrew** [comment]
@AmeliaBR I don't think we specifically have an issue for multicol, there's mention of fragment styling in [overflow-4](https://www.w3.org/TR/css-overflow-4/#fragment-styling) which would be one way to achieve this. It's definitely something people ask for.

**Pull request merged**